Respect policy list expiration

Snawoot opened the pull request:
New pull request since I can't change branch content in not mine repo (and can't change PR source branch either).

Changes since last attempt:
* Tests for parent generator class also are wrapped by context manager introduced by previous versions of this PR. It resolves dependency of test on policy.json file in repo.
* Parameterizing code now is common test util (and quite easy to use I hope). So it is not related only to config tests and I used general naming.
* Naming improved

Signed-off-by: Vladislav Yarmak <vladislav@vm-0.com>

Snawoot commented:
This PR should resolve #1 and #3

Snawoot commented:
UPD: made linter happy

sydneyli commented on starttls_policy/tests/util.py:
```diff
@@ -0,0 +1,18 @@
+        tname, args = entry[0], entry[1:]
+        test_name = "test_" + tname
+        wrapped_test = test(*args)
+        setattr(cls, test_name, wrapped_test)
```
Rather than have an API that requires two calls (one to `partial` the args except for the class instance, and one to bind the functions to the class) I would prefer the API to perform a single call.

I'm not sure how you could do this with a single decorator, but maybe we could do it with just `parametrize_over`. That is, would it be possible to define `config_test` as a member function of `TestPostfixGenerator` and have`parametrize_over` apply the partial args?

Something like the below might work, as long as `test` is bound to an instance of `cls`.
```suggestion
        def _partial(func, *args, **kwargs):
          def wrapped(self):
            return func(self, *args, **kwargs)
          return wrapped
        setattr(cls, test_name, _partial(test, *args))
```

Snawoot replied:
fixed

sydneyli commented on starttls_policy/tests/util.py:
```diff
@@ -0,0 +1,18 @@
+"""Various utils used by tests"""
+
+def parameterized(func):
```
this is really not related at all but TIL that parameterize [can be spelled](https://en.wikipedia.org/wiki/Parametrization) "parametrize", and also with an s instead of a z. So there's 4 different spellings that are all valid!

Snawoot replied:
Switched to pytest variant of spelling

Snawoot commented:
Changes made:
* Moved parameterized method into class body.
* Removed decorator and made function `parametrize_over` decorate method internally.
* Introduced type to construct and contain parameterized test args.

Snawoot commented:
Ok, thank you!
